assets/app/scripts/controllers/projects.js

```diff
@@ -26,46 +28,40 @@ angular.module('openshiftConsole')
       $('#openshift-logo').off('click.projectsPage');
     });
 
-    $scope.toggleConfirm = function(project) {
-      // toggle prompt for user to decide if they are sure about deletion
-      var projectName = project.metadata.name;
+    $scope.openDeleteModal = function(project) {
+      // opening the modal with settings scope as parent
+      var modalInstance = $modal.open({
+        animation: true,
+        templateUrl: 'views/modals/delete-project.html',
+        controller: 'DeleteModalController'
+      });
 
-      // disable the trash button if current project is being deleted
-      if (!$scope.deletingMap[projectName]) {
-        $scope.confirmingMap[projectName] = !$scope.confirmingMap[projectName];
-      }
-    };
+      modalInstance.result.then(function() {
+        // actually deleting the project
+        var projectName = project.metadata.name;
+        delete $scope.alerts[projectName];
+        DataService.delete('projects', projectName, $scope)
+        .then(function() {
+          // called if successful deletion
+          $scope.alerts[projectName] = {
+            type: "success",
+            message: "Project " + $filter('displayName')(project) + " was marked for deletion."
+          };
 
-    $scope.deleteProject = function(project) {
-      // actually deleting the project
-      var projectName = project.metadata.name;
-      delete $scope.alerts[projectName];
-      $scope.confirmingMap[projectName] = false;
-      $scope.deletingMap[projectName] = true;
-      DataService.delete('projects', projectName, $scope)
-      .then(function() {
-        // called if successful deletion
-        $scope.alerts[projectName] = {
-          type: "success",
-          message: "Project " + $filter('displayName')(project) + " was successfully deleted."
-        };
-        delete $scope.projects[projectName];
-      })
-      .catch(function(err) {
-        // called if failure to delete
-        $scope.alerts[projectName] = {
-          type: "error",
-          message: "Project " + $filter('displayName')(project) + " could not be deleted.",
-          details: err
-        };
-        Logger.error("Project " + $filter('displayName')(project) + " could not be deleted.", err);
-      })
-      .finally(function() {
-        // common stuff
-        $scope.deletingMap[projectName] = false;
-        $timeout(function() {
-          delete $scope.alerts[projectName];
-        }, 10000);
+          DataService.list("projects", $scope, function(projects) {
```

Comment:
pull these 4 lines out into a function called loadProjects and call it from here and from within the AuthService.withUser block so that we arent duplicating this